@xen-orchestra/web/src/pages/vm/new.vue

```diff
@@ -762,16 +765,56 @@ const redirectToHome = () => {
   router.push({ name: '/' })
 }
 
-const vmData = computed(() => {
-  const optionalFields = {
-    ...(vmState.affinity_host && { affinity: vmState.affinity_host }),
-    ...(installMode.method && installMode.method !== 'no-config' && { install: installMode }),
-    ...(installMode.method === 'custom_config' && {
-      ...(vmState.cloudConfig && { cloud_config: vmState.cloudConfig }),
-      ...(vmState.networkConfig && { network_config: vmState.networkConfig }),
-    }),
+function getExistingVdisDiff(vdi1: Vdi, vdi2: Vdi) {
+  const changes: Record<string, unknown> = {}
+  let hasChanged = false
+
+  for (const _key in vdi1) {
+    const key = _key as keyof Vdi
+
+    if (vdi1[key] !== vdi2[key]) {
+      hasChanged = true
+      changes[key] = vdi2[key]
+    }
   }
-  const templateVifs = vmState.new_vm_template.VIFs
+
+  return hasChanged ? (changes as Partial<Vdi>) : undefined
+}
+
+const modifiedExistingVdis = computed(() => {
+  return vmState.existingVdis.reduce((acc, vdi, index) => {
+    const defaultVdi = defaultExistingVdis.value[index]
+    const changes = getExistingVdisDiff(defaultVdi, vdi)
+
+    if (changes) {
+      acc.push({ ...changes, userdevice: vdi.userdevice })
+    }
+
+    return acc
+  }, [] as Partial<Vdi>[])
+})
+
+const vmData = computed(() => {
+  const vdisToSend = [...vmState.vdis, ...modifiedExistingVdis.value].map(vdi => ({
+    ...vdi,
+    ...(vdi.size && { size: giBToBytes(vdi.size) }),
+  }))
+
+  const optionalFields = Object.assign(
+    {},
+    vdisToSend.length > 0 && { vdis: vdisToSend },
+    vmState.affinity_host && { affinity: vmState.affinity_host },
+    vmState.installMode !== 'no-config' && {
+      install: { method: vmState.installMode, repository: vmState.selectedVdi },
```

Comment:
As reported by @S3bastianCZ, in case of `installMode === "PXE"`, repository must be an empty string.